src/modules/messagePreview/module.js

```diff
@@ -25,27 +30,43 @@ subscriptions.set('messageCreate', async (client, message) => {
         if (
             !msg.channel
                 .permissionsFor(message.author)
-                ?.has([Permissions.FLAGS.VIEW_CHANNEL, Permissions.FLAGS.READ_MESSAGE_HISTORY])
+                ?.has([
+                    PermissionsBitField.Flags.ViewChannel,
+                    PermissionsBitField.Flags.ReadMessageHistory,
+                ])
         )
             return;
         if (!msg.member) await msg.guild.members.fetch(msg.author);
 
-        const embed = new MessageEmbed()
-            .setAuthor(msg.member.displayName, msg.member.displayAvatarURL({ dynamic: true }))
-            .setDescription(msg.content)
+        const embed = new EmbedBuilder()
+            .setAuthor({
+                name: msg.member.displayName,
+                iconURL: msg.member.displayAvatarURL(),
+                url: msg.url,
+            })
             .setColor(msg.member.displayColor)
-            .setFooter(`Requested by ${message.member.displayName}`)
-            .setTimestamp(msg.createdTimestamp);
+            .setFooter({ text: `Requested by ${message.member.displayName}` })
+            .setTimestamp(msg.createdTimestamp)
+            .setURL(msg.url);
 
+        const additionalEmbeds = [];
+        if (msg.content) embed.setDescription(msg.content);
         if (msg.attachments.size) {
```

Comment:
you should scrap the message if both conditions are false (i.e. an empty message) thouh i have no idea how one could even link such message

Reply:
This can happen if a user sends a sticker, but I think otherwise, its not a big deal

Reply:
so you accept empty embed?

Reply:
Yeah. Its a weird edge case, and lets users know somethign happened there. Plus it would still have an author field

Reply:
System messages also cause this. I've made it not crash, but it should be fine, as we don't get many system messages in channels we are moving from.